### DISCUSSION TITLE
Tablet zoom

### DISCUSSION THREAD

**Discussion by TomStark1** [sumatrapdfreader/sumatrapdf#4604] [Ideas]
Hello Sumatra PDF team,

I've encountered an issue while using Sumatra PDF on a touch device. When I try to zoom into a PDF document using a two-finger gesture, the view always zooms into the top-left corner of the document instead of the area between my fingers. This makes it difficult to zoom in on other areas of the page.

Steps to reproduce:

Open a PDF in Sumatra PDF on a touch-enabled device.
Use a two-finger zoom gesture to zoom into a specific area of the document.
Expected behavior:
The zoom should focus on the area between the two fingers.

Observed behavior:
The zoom always centers on the top-left corner of the document.

Is there a known fix for this, or is there a plan to address it? Thanks for your help and all your hard work!

**Comment by GitHubRulesOK** [comment]
That was always the issue with zoom it is top left origin unless object focused
but seem to think an attempt was more recently (MAY 2024) made to adjust the touch behaviour
see https://github.com/sumatrapdfreader/sumatrapdf/commit/485817ee9eb392e1f1b831c72ab936ec30d07d98
so are you testing with latest pre-release (August 2024) ?

**Reply by TomStark1** [reply]
Ah, yes, I just tried it, and it kinda works, but the zoom only adjusts once I release my fingers. While zooming, there’s no change in the zoom level—it just stays the same until I let go. I’ll wait for the next release and see if it improves. :)

Thanks for the quick response!

**Reply by GitHubRulesOK** [reply]
It wont be any better in a release than pre-release unless somebody changes any functionality
 so you need to be more descriptive as to what needs adjust as many devices may behave differently.
